docker-compose-local.yml

```diff
@@ -5,39 +5,30 @@ services:
     container_name: simulator-server
     environment:
       - PORT=1212
-      - KUBE_SCHEDULER_SIMULATOR_ETCD_URL=http://simulator-etcd:2379
-      - CORS_ALLOWED_ORIGIN_LIST=http://${SIMULATOR_EXTERNAL_IP:-localhost}:3000
+      - KUBE_SCHEDULER_SIMULATOR_ETCD_URL=http://127.0.0.1:2379
       - KUBE_API_HOST=0.0.0.0
       - KUBE_API_PORT=3131
     ports:
       - "1212:1212"
+      - "1313:1313"
       - "3131:3131"
     restart: always
     tty: true
     networks:
       - simulator-internal-network
+    volumes:
+      - ./kwok.yaml:/.kwok/kwok.yaml
   simulator-frontend:
     image: simulator-frontend
     restart: always
     container_name: simulator-frontend
     environment:
       - HOST=0.0.0.0
       - BASE_URL=http://${SIMULATOR_EXTERNAL_IP:-localhost}:1212
-      - KUBE_API_SERVER_URL=http://${SIMULATOR_EXTERNAL_IP:-localhost}:3131
+      - KUBE_API_SERVER_URL=http://${SIMULATOR_EXTERNAL_IP:-localhost}:1313
     ports:
     - "3000:3000"
     tty: true
-  simulator-etcd:
-    image: quay.io/coreos/etcd:v3.4.26
-    container_name: simulator-etcd
-    restart: always
-    volumes:
-      - simulator-etcd-data:/var/lib/etcd
-    command: etcd --advertise-client-urls http://simulator-etcd:2379 --data-dir /var/lib/etcd --listen-client-urls http://0.0.0.0:2379 --initial-cluster-state new --initial-cluster-token tkn
-    networks:
-      - simulator-internal-network
-volumes:
-  simulator-etcd-data:
```

Comment:
I think it's better to create clusters directly from existing mirrors, so that the simulator doesn't need to be concerned with the creation and deletion of clusters at all, and only needs to be externally supplied with etcd ports.


``` yaml
  simulator-cluster:
    image: registry.k8s.io/kwok/cluster:v0.5.0-k8s.v1.29.0
    container_name: simulator-cluster
    restart: always
    ports:
      - "8080:8080"
    environment:
      - KWOK_KUBE_APISERVER_PORT=8080
    volumes:
      - simulator-etcd-data:/var/lib/etcd
      - ./kwokctl.yaml:/root/.kwok/kwok.yaml
```

```yaml
# kwokctl.yaml
kind: KwokctlConfiguration
apiVersion: config.kwok.x-k8s.io/v1alpha1
options:
  etcdPort: 2379
  etcdPrefix: /kube-scheduler-simulator
  disableKubeScheduler: true
componentsPatches:
- name: kube-apiserver
  extraArgs:
  - key: cors-allowed-origins
    value: ^*$
```

Reply:
How can we know if cluster is ready ?

Reply:
https://t.co/VGv4GwKzmm

Sorry for the clutter. Although I tried to fix these codes, it couldn't catch the watchResources response. it's suspended and I don't have no idea why it's caused. Do you have some idea ?